pyanaconda/modules/timezone/timezone_interface.py

```diff
@@ -0,0 +1,108 @@
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+
+from pyanaconda.dbus.constants import MODULE_TIMEZONE_NAME
+from pyanaconda.dbus.property import emits_properties_changed
+from pyanaconda.dbus.typing import *  # pylint: disable=wildcard-import
+from pyanaconda.modules.base_interface import KickstartModuleInterface
+from pyanaconda.dbus.interface import dbus_interface
+
+
+@dbus_interface(MODULE_TIMEZONE_NAME)
+class TimezoneInterface(KickstartModuleInterface):
+    """DBus interface for Timezone module."""
+
+    def connect_signals(self):
+        super().connect_signals()
+        self.implementation.timezone_changed.connect(self.changed("Timezone"))
+        self.implementation.is_utc_changed.connect(self.changed("IsUTC"))
+        self.implementation.ntp_enabled_changed.connect(self.changed("NTPEnabled"))
+
+    @property
+    def Timezone(self) -> Str:
+        """Timezone the system will use."""
+        return self.implementation.timezone
+
+    @emits_properties_changed
+    def SetTimezone(self, timezone: Str):
+        """Set the timezone.
+
+        Sets the system time zone to timezone. To view a list of
+        available time zones, use the timedatectl list-timezones
+        command.
+
+        Example: Europe/Prague
+
+        :param timezone: a string with a timezone
+        """
+        self.implementation.set_timezone(timezone)
+
+    @property
+    def IsUTC(self) -> Bool:
+        """Is the hardware clock set to UTC?
+
+        The system assumes that the hardware clock is set to UTC
+        (Greenwich Mean) time, if true.
+
+        :return: True, if the hardware clock set to UTC, otherwise False
+        """
+        return self.implementation.is_utc
+
+    @emits_properties_changed
+    def SetUTC(self, is_utc: Bool):
+        """Set if the hardware clock set to UTC or not.
+
+        :param is_utc: Is the hardware clock set to UTC?
+        """
+        self.implementation.set_utc(is_utc)
+
+    @property
+    def NTPEnabled(self) -> Bool:
+        """Is automatic starting of NTP service enabled?
+
+        :return: True, if the service is enabled, otherwise false.
+        """
+        return self.implementation.ntp_enabled
+
+    @emits_properties_changed
+    def SetNTP(self, ntp_enabled: Bool):
```

Comment:
This is not `SetNTP` but rather `EnableNTP` or `ToggleNTP` or if you want just the same logic then `SetNTPEnabled`. I personally see set ntp as set ntp servers.

Reply:
Fixed.

Reply:
`SetNTPEnabled` enabled looks best to me personally. Also for future use cases it might be good to ask someone who maintains a DBUS API (udisks/cockpit) how they handle Boolean values - set or toggle (I would guess set is preffered).